src/components/general/NotificationsSideSheet/index.tsx

```diff
@@ -78,6 +52,70 @@ const NotificationsSideSheet: React.FC<NotificationsSideSheetProps> = ({
         [markNotificationsAsSeenAsync]
     );
 
+    const onReadAll = React.useCallback(() =>
+        unReadNotifications.length > 0 &&
+        markNotificationAsSeen(unReadNotifications),
+
+        [unReadNotifications, markNotificationAsSeen, isMarkingNotifications]);
+
+    return (
+        <div className={styles.notificationsContainer}>
+            <Tabs activeTabKey={activeTabKey} onChange={changeTabKey}>
+                <Tab tabKey={"unread"} title={<div> Unread {chip()}</div>}>
+                    <div className={styles.tabContent}>
+                        {unReadNotifications.length === 0 &&
+                            !(isFetchingUnReadNotifications) ? (
+                                <div className={styles.noNotificationMessage}>
+                                    <span>You don't have any unread notifications</span>
+                                    </div>
+                            ) : (isFetchingUnReadNotifications ? (
+                                <Spinner centered />
+                            ) : (
+                                    <>
+                                        <Button
+                                            onClick={onReadAll}
+                                            disabled={unReadNotifications.length <= 0}
+                                        >
+                                            {isMarkingNotifications ? <Spinner inline /> : 'Mark all as read'}
+                                        </Button>
+                                    <NotificationDateDivisions notifications={unReadNotifications} />
+                                    </>
+                                ))}
+                    </div>
```

Comment:
too much logic inside render method, either refactor to callback/memo or create a new component(s)

Reply:
Removed some logic from the render method and refactored to memo